packages/backend/src/backend_factory.ts

```diff
@@ -121,5 +155,19 @@ export const defineBackend = <T extends DefineBackendProps>(
   return {
     ...backend.resources,
     createStack: backend.createStack,
+    setCustomOutput: backend.setCustomOutput,
+    addToClientConfig: (clientConfigPart: Partial<ClientConfig>) => {
+      // TODO This is alternative proposal
+      // NO-OP for now but the idea is that we allow to put a partial of client config schema
+      // into some output
+      // and just merge it with client config at read.
+
+      // This likely could use client config unification.
+      // I.e. backend def -> unified client config partial to outputs -> deployment ->
+      // -> outputs -> read unified client config -> map to js/ts/dart/whatever.
+
+      // Should we start crafting unified schema nad return it from generateClientConfig API ?
+      // and map to old/v6 formats in generateClientConfigToFile ?
+    }
```

Comment:
The disadvantage that `setCustomOutput` with that `path` pointers to config has is that it's hard to figure out what paths should be used.

This presents an alternative approach where we unify client config schema and use it to provide typings.